debug/handler/interceptorTxDebug.go

```diff
@@ -0,0 +1,91 @@
+package handler
+
+import (
+	"encoding/hex"
+	"sync"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+
+	p2p2 "github.com/multiversx/mx-chain-communication-go/p2p"
+	"github.com/multiversx/mx-chain-go/p2p"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+const (
+	interceptedTx         = "intercepted tx"
+	interceptedRewardTx   = "intercepted reward tx"
+	interceptedUnsignedTx = "intercepted unsigned tx"
+)
+
+type receivedTxEvent struct {
+	from                  string
+	originator            string
+	firstTimeReceivedMile int64
+	numReceived           int
+	txType                string
+	// TODO check what fields to add
+}
+
+type interceptorTxDebug struct {
+	mutex                sync.RWMutex
+	receivedTxsBroadcast map[string]*receivedTxEvent
+}
+
+func NewInterceptorTxDebug() *interceptorTxDebug {
+	return &interceptorTxDebug{
+		receivedTxsBroadcast: make(map[string]*receivedTxEvent),
+	}
+}
+
+func (id *interceptorTxDebug) Process(data process.InterceptedData, msg p2p.MessageP2P, fromConnectedPeer core.PeerID) {
+	if msg.BroadcastMethod() != p2p2.Broadcast {
+		return
+	}
+
+	isCorrectType := data.Type() == interceptedTx // && data.Type() == interceptedRewardTx && data.Type() == interceptedUnsignedTx
+	if !isCorrectType {
+		return
+	}
+
+	id.mutex.Lock()
+	defer id.mutex.Unlock()
+
+	receivedTxE, found := id.receivedTxsBroadcast[string(data.Hash())]
+	if !found {
+		hexHash := hex.EncodeToString(data.Hash())
+		id.receivedTxsBroadcast[hexHash] = &receivedTxEvent{
+			originator:            core.PeerID(msg.From()).Pretty(),
+			from:                  fromConnectedPeer.Pretty(),
+			numReceived:           1,
+			firstTimeReceivedMile: getCurrentTimeStampMilli(),
+			txType:                data.Type(),
+		}
+		return
+	}
+
+	receivedTxE.numReceived++
```

Comment:
I think here you should update the from field as well, in order to properly save where the message came from? 🤔